### DISCUSSION TITLE
umi4 react-query 如何将某一个请求的缓存去掉，比如退出登录以后，取消用户信息这个接口的缓存

### DISCUSSION THREAD

**Discussion by daisybaicai** [umijs/umi#11355] [Q&A]
umi4 react-query 如何将某一个请求的缓存去掉，比如退出登录以后，取消用户信息这个接口的缓存

**Comment by fz6m** [comment]
react-query 的缓存是根据你的 key 变化来的，比如一个对象作为 key ，当这个对象键值对发生改变后，就会重新获取，所以你可以把查询的筛选参数作为 key 。

如果同样的 key ，但数据是在不断变化的，可以调用 `useQuery().refetch()`
